frontend/src/concepts/pipelines/content/tables/usePipelineRunVersionInfo.ts

```diff
@@ -0,0 +1,21 @@
+import * as React from 'react';
+import { getPipelineVersionResourceRef } from '~/concepts/pipelines/content/tables/utils';
+import { PipelineRunJobKF, PipelineRunKF, PipelineVersionKF } from '~/concepts/pipelines/kfTypes';
+import { PipelineRunVersionsContext } from '~/pages/pipelines/global/runs/PipelineRunVersionsContext';
+
+const usePipelineRunVersionInfo = (
+  resource: PipelineRunKF | PipelineRunJobKF | null,
+): {
+  version?: PipelineVersionKF;
```

Comment:
Don't we always return version and error, they're just potentially undefined/null? We only want to make props optional if they're truly optional, not just to account for undefined values.